src/DataFixtures/PackageFixtures.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace App\DataFixtures;
+
+use App\Entity\Package;
+use App\Entity\User;
+use App\Model\ProviderManager;
+use Doctrine\Bundle\FixturesBundle\Fixture;
+use Doctrine\Common\DataFixtures\DependentFixtureInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ObjectManager;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Output\ConsoleOutput;
+
+class PackageFixtures extends Fixture implements DependentFixtureInterface
+{
+    private EntityManagerInterface $em;
+
+    private ProviderManager $providerManager;
+
+    public function __construct(EntityManagerInterface $em, ProviderManager $providerManager)
+    {
+        $this->em = $em;
+        $this->providerManager = $providerManager;
+    }
+
+    public function getDependencies()
+    {
+        return [
+            UserFixtures::class,
+        ];
+    }
+
+    public function load(ObjectManager $manager)
+    {
+        $output = new ConsoleOutput();
+
+        $urls = $this->getPackageUrls();
+
+        $progressBar = new ProgressBar($output, count($urls));
+        $progressBar->setFormat('%current%/%max% [%bar%] %percent:3s%% (%remaining% left) %message%');
+
+        $progressBar->setMessage('');
+        $progressBar->start();
+
+        /** @var User $maintainer */
+        $maintainer = $this->getReference(UserFixtures::PACKAGE_MAINTAINER_REFERENCE);
+
+        foreach ($urls as $url) {
+            $package = new Package;
+
+            $package->addMaintainer($maintainer);
+            $package->setRepository($url);
+
+            $manager->persist($package);
+            $manager->flush();
+
+            $this->providerManager->insertPackage($package);
+
+            $progressBar->advance();
+            $progressBar->setMessage($url);
```

Comment:
shouldn't this message be set when starting to process the URL rather than when we're done with it ?

Reply:
Good point. I had to manually call `->display()` after `setMessage()` though, or the message would not display before `advance()` anyway!